src/compute/src/server.rs

```diff
@@ -156,6 +174,43 @@ pub fn serve(
 type CommandReceiver = crossbeam_channel::Receiver<ComputeCommand>;
 type ResponseSender = mpsc::UnboundedSender<ComputeResponse>;
 
+struct CommandReceiverQueue {
+    queue: Rc<RefCell<VecDeque<Result<ComputeCommand, TryRecvError>>>>,
+}
+
+impl CommandReceiverQueue {
+    fn try_recv(&mut self) -> Result<ComputeCommand, TryRecvError> {
+        let mut b = (*self.queue).borrow_mut();
+        match b.pop_front() {
+            Some(Ok(cmd)) => Ok(cmd),
+            Some(Err(e)) => Err(e),
+            None => Err(TryRecvError::Empty),
+        }
+    }
+
+    fn recv<A: Allocate>(&mut self, worker: &mut Worker<A>) -> Result<ComputeCommand, RecvError> {
+        // This takes the worker, so it can step timely while no result
+        // is available.
```

Comment:
I wonder if there's an alternative design where we broadcast the initialization commands to all workers and only unicast once the initialization is done? This might have the downside of added complexity and making it harder to recover from errors during initialization.

Reply:
I think that's a possibility. Also regarding #14379 . It might make the cloud deployment easier if computed's start as "empty shell" and receive their configuration in the CreateInstance. If we do this, broadcasting the CreateInstance is a requirement.